apps/client/src/common/components/client-modal/RedirectClientModal.tsx

```diff
@@ -1,64 +1,128 @@
 import { useState } from 'react';
 import {
-  Button,
+  IconButton,
   Input,
-  InputGroup,
-  InputLeftAddon,
   Modal,
   ModalBody,
   ModalCloseButton,
   ModalContent,
-  ModalFooter,
   ModalHeader,
   ModalOverlay,
+  Select,
 } from '@chakra-ui/react';
+import { IoArrowForward } from '@react-icons/all-files/io5/IoArrowForward';
 
 import { setClientRemote } from '../../hooks/useSocket';
+import useUrlPresets from '../../hooks-query/useUrlPresets';
+import Info from '../info/Info';
+
+import style from './RedirectClientModal.module.scss';
 
 interface RedirectClientModalProps {
   id: string;
-  name?: string;
-  path?: string;
+  name: string;
+  currentPath: string;
+  origin: string;
   isOpen: boolean;
   onClose: () => void;
 }
 
 export function RedirectClientModal(props: RedirectClientModalProps) {
-  const { id, isOpen, name = '', path: currentPath = '', onClose } = props;
+  const { id, isOpen, name, currentPath, origin, onClose } = props;
+  const { data } = useUrlPresets();
   const [path, setPath] = useState(currentPath);
+  const [selected, setSelected] = useState('/');
 
   const { setRedirect } = setClientRemote;
 
-  const handleRedirect = () => {
-    if (path !== currentPath && path !== '') {
-      setRedirect({ target: id, redirect: path });
+  const handleRedirect = (newPath: string) => {
+    if (newPath === '/' || newPath === currentPath) {
+      return;
     }
+    setRedirect({ target: id, redirect: newPath });
     onClose();
   };
 
-  const host = window.location.origin;
-  const canSubmit = path !== currentPath && path !== '';
+  const enabledPresets = data.filter((preset) => preset.enabled);
 
   return (
     <Modal isOpen={isOpen} onClose={onClose} variant='ontime'>
       <ModalOverlay />
-      <ModalContent>
+      <ModalContent maxWidth='max(480px, 35vw)'>
         <ModalHeader>Redirect: {name}</ModalHeader>
         <ModalCloseButton />
         <ModalBody>
-          <InputGroup variant='ontime-filled' size='md'>
-            <InputLeftAddon>{host}</InputLeftAddon>
-            <Input placeholder='minimal?key=0000ffff' value={path} onChange={(event) => setPath(event.target.value)} />
-          </InputGroup>
+          <Info>
+            Remotely redirect the client to a different URL. <br />
+            Either by selecting a URL Preset or entering a custom path.
+            <br />
+            <br />
+            <a href='/editor?settings=feature_settings__urlpresets' target='_blank' className={style.link}>
+              Manage URL Presets
+            </a>
+          </Info>
+          <div>
+            <span className={style.label}>Select View or URL Preset</span>
+            <div className={style.textEntry}>
+              <Select
+                size='md'
+                variant='ontime'
+                isDisabled={enabledPresets.length === 0}
+                onChange={(event) => setSelected(`/${event.target.value}`)}
```

Comment:
```suggestion
                onChange={(event) => setSelected(`${event.target.value}`)}
```

this was resulting in double slashes

Reply:
good catch, I have fixed the issue